python/tests/integration/arcticdb/test_read_batch_more.py

```diff
@@ -0,0 +1,364 @@
+    df1_all = pd.concat([df1_till1, df1_2],ignore_index=True)
+    df1_all.reset_index(inplace = True, drop = True)
+    metadata1 = {"version" : 1 , "data" : [1,3,5]}
+
+    symbol2 = "s2"
+    df2_0 = create_df_index_datetime(5, 0, 10)
+    df2_1 = create_df_index_datetime(5, 10, 20)
+    df2_all = pd.concat([df2_0, df2_1])
+    df2_all_added = df2_all.copy(deep=True)
+    df2_all_added["ADDED"] = df2_all_added["COL_1"] + df2_all_added["COL_2"] + 1
+    metadata2 = {"Version" : 1.23 , "data" : {"a": 1, "b": 3,"c": 5}}
+    metadata3 = {"final" : [1, 2]}
+
+    lib.write(symbol1, df1_0, metadata=metadata1)
+    lib.append(symbol1, df1_1)
+    lib.append(symbol1, df1_2)
+
+    lib.write(symbol2, df2_0)
+    lib.write_metadata(symbol2, metadata2)
+    lib.append(symbol2, df2_1, metadata3)
+
+    # Simplest Boolean condition in query
+    qdf1 = "bool == True"
+    q1 = QueryBuilder()
+    q1 = q1[q1["bool"]]
+    # Boolean AND Integer condition in query
+    qdf2 = "bool == True and int8 > 5" 
+    q2 = QueryBuilder()
+    q2 = q2[q2["bool"] & (q2["int8"] > 5)]
+    qdf3 = "COL_1 > COL_2"
+    # With Apply clause
+    q3 = QueryBuilder()
+    q3 = q3[q3["COL_1"] > q3["COL_2"]]
+    q3.apply("ADDED", q3["COL_1"] + q3["COL_2"] + 1)
+    # Text and float clause in query
+    qdf4 = "short == 'K' and float64 > 12.5"
+    q4 = QueryBuilder()
+    q4 = q4[(q4["short"] == 'K') & (q4["float64"] > 12.5)]
+
+    batch = lib.read_batch(symbols=[symbol1,
+                                    ReadRequest(symbol1, as_of=0), 
+                                    ReadRequest(symbol1, query_builder=q1, as_of=0), 
+                                    symbol2,
+                                    ReadRequest(symbol1, query_builder=q2),
+                                    ReadRequest(symbol2, query_builder=q3),
+                                    ReadRequest(symbol2, as_of=0),
+                                    ReadRequest(symbol1, query_builder=q4)
+                                    ])
+
+    assert_frame_equal(df1_all, batch[0].data)
+    assert None == batch[0].metadata #metadata is only per the version it was specified for
+    assert_frame_equal(df1_0, batch[1].data)
+    # Filter with boolean condition
+    dfqapplied = df1_0.query(qdf1)
+    assert_frame_equal_row_range_fix(dfqapplied, batch[2].data)
+    assert_frame_equal(df2_all, batch[3].data)
+    # Filter with boolean and integer condition
+    dfqapplied = df1_all.query(qdf2)
+    assert_frame_equal_row_range_fix(dfqapplied, batch[4].data)
+    # filter with column between another column and apply clause
+    dfqapplied = df2_all_added.query(qdf3)
+    assert_frame_equal_row_range_fix(dfqapplied, batch[5].data)
+    assert metadata3 == batch[5].metadata
+    assert metadata3 == lib.read_metadata(symbol2).metadata
+
+    # Last test will fail on Pandas 1 due to fact that
+    # when empty df is returned object type columns will have type float64
+    v = pd.__version__
```

Comment:
There is a constant IS_PANDAS_TWO that we should use instead of this. It's defined in arctic and use in several places.